opus/import/populate_obs_instrument_VGRSS_prof.py

```diff
@@ -0,0 +1,555 @@
+# 0 (parallel to north pole) to 90 (parallel to ring)
+# We would like to use emission angle to get both min/max of incidence angle.
+# Emission angle is 90-180 (dark side), so incidence angle is 180 - emission
+# angle.
+def populate_obs_ring_geometry_VGRSS_incidence1_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    inc = index_row['INCIDENCE_ANGLE']
+
+    max_ea = index_row['MAXIMUM_EMISSION_ANGLE']
+    cal_inc = 180 - max_ea
+    msg = ('The difference between incidence angle and 180 - emission is' +
+           ' more than 0.005, volume: VG_2803.')
+    assert abs(cal_inc - inc) <= 0.005, msg
+    return 180. - max_ea
+
+def populate_obs_ring_geometry_VGRSS_incidence2_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    inc = index_row['INCIDENCE_ANGLE']
+
+    min_ea = index_row['MINIMUM_EMISSION_ANGLE']
+    cal_inc = 180 - min_ea
+    msg = ('The difference between incidence angle and 180 - emission is' +
+           ' more than 0.005, volume: VG_2803.')
+    assert abs(cal_inc - inc) <= 0.005, msg
+    return 180. - min_ea
+
+# North based inc: the angle between the point where incoming source photons hit
+# the ring to the normal vector on the NORTH side of the ring. 0-90 when north
+# side of the ring is lit, and 90-180 when south side is lit.
+# Since south side is lit, north based incidence angle is between 90-180.
+# It's 180 - inc, which will be the same as emission angle.
+def populate_obs_ring_geometry_VGRSS_north_based_incidence1_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    min_ea = index_row['MINIMUM_EMISSION_ANGLE']
+
+    return min_ea
+
+def populate_obs_ring_geometry_VGRSS_north_based_incidence2_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    max_ea = index_row['MAXIMUM_EMISSION_ANGLE']
+
+    return max_ea
+
+# Emission angle: the angle between the normal vector on the LIT side, to the
+# direction where outgoing photons to the observer. 0-90 when observer is at the
+# lit side of the ring, and 90-180 when it's at the dark side.
+# Since observer is at the dark side, ea is between 90-180
+def populate_obs_ring_geometry_VGRSS_emission1_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    min_ea = index_row['MINIMUM_EMISSION_ANGLE']
+
+    return min_ea
+
+def populate_obs_ring_geometry_VGRSS_emission2_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    max_ea = index_row['MAXIMUM_EMISSION_ANGLE']
+
+    return max_ea
+
+# North based ea: the angle between the normal vector on the NORTH side of the
+# ring, to the direction where outgoing photons to the observer. 0-90 when
+# observer is at the north side of the ring, and 90-180 when it's at the south
+# side.
+# Since observer is at north, north based emission angle is between 0-90.
```

Comment:
Isn't the observer at the south?

Reply:
Based on slack discussion, the observer is at north.